GeometryCore/src/Polygon.cpp

```diff
@@ -144,13 +142,13 @@ namespace geom_utils
 
         // Build lower hull
         for (size_t i = 0; i < pointsCopy.size(); ++i) {
-            while (k >= 2 && my_cross(convexHull.points[k - 2], convexHull.points[k - 1], pointsCopy[i]) <= 0) k--;
+            while (k >= 2 && geom_utils::cross(FPoint2D(convexHull.points[k - 2] - pointsCopy[i]), FPoint2D(convexHull.points[k - 1] - pointsCopy[i])) <= 0) k--;
```

Comment:
Please check carefully if this new version is identical to the previous cross(OA, OB) vectors... at the first glance, looks like you have wrong cross product

Reply:
Yes, it is my mistake. I will fix it